.github/workflows/cmake.yml

```diff
@@ -60,17 +67,29 @@ jobs:
         with:
           os: 'Serenity'
           arch: ${{ matrix.arch }}
-          toolchain: 'GNU'
+          toolchain: ${{ matrix.toolchain }}
           cache_key_extra: ${{ matrix.debug-options }}
           serenity_ccache_path: ${{ env.SERENITY_CCACHE_DIR }}
           toolchain_ccache_path: ${{ env.TOOLCHAIN_CCACHE_DIR }}
           download_cache_path: ${{ github.workspace }}/Build/caches
 
+      - name: Assign Build Parameters
+        id: 'build-parameters'
+        run: |
+          if ${{ matrix.toolchain == 'Clang' }} ; then
+            echo "build_directory=${{ github.workspace }}/Build/${{ matrix.arch }}clang" >> "$GITHUB_OUTPUT"
+          elif ${{ matrix.toolchain == 'GNU' }} ; then
+            echo "build_directory=${{ github.workspace }}/Build/${{ matrix.arch }}" >> "$GITHUB_OUTPUT"
+          fi
+
       - name: Build toolchain
         if: ${{ !steps.cache-restore.outputs.toolchain_prebuilt_hit }}
-        run: ARCH="${{ matrix.arch }}" ${{ github.workspace }}/Toolchain/BuildGNU.sh
+        run: |
+          ${{ github.workspace }}/Toolchain/Build${{ matrix.toolchain }}.sh --ci
         env:
+          ARCH: ${{ matrix.arch}}
           CCACHE_DIR: ${{ env.TOOLCHAIN_CCACHE_DIR }}
+          TRY_USE_LOCAL_TOOLCHAIN: 'y'
```

Comment:
2 improvement ideas:

We should(?) Remove the TRY_USE_LOCAL_TOOLCHAIN from BuildClang.sh now that we're using the caching strat from GA. 

as far as i can tell, BuildGNU.sh doesn't use ccache.

Reply:
Yep exactly - wanted to do that in a follow up though because this PR got big enough :sweat_smile: